File: python/sliding_window/LC_1695_MaximumErasureValue.py
# https://leetcode.com/problems/maximum-erasure-value
from typing import List


class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        sub_array = set()
        left = right = curr_sum = max_sum = 0
        while right < len(nums):
            if nums[right] not in sub_array:
                sub_array.add(nums[right])
                curr_sum += nums[right]
                max_sum = max(max_sum, curr_sum)
                right += 1
            else:
                sub_array.remove(nums[left])
                curr_sum -= nums[left]
                left += 1
        return max_sum

File: java/src/sliding_window/LC_1695_MaximumErasureValue.java
package sliding_window;
// https://leetcode.com/problems/maximum-erasure-value

import java.util.HashSet;

public class LC_1695_MaximumErasureValue {
    public int maximumUniqueSubarray(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int left = 0, max_sum = 0, current_sum = 0;
        for (int num : nums) {
            while (set.contains(num)) {
                set.remove(nums[left]);
                current_sum -= nums[left];
                left++;
            }
            set.add(num);
            current_sum += num;
            max_sum = Math.max(max_sum, current_sum);
        }
        return max_sum;
    }
}
